src/Nethermind/Nethermind.TxPool/TxBroadcaster.cs

```diff
@@ -97,31 +97,35 @@ internal class TxBroadcaster : IDisposable
         // only for testing reasons
         internal Transaction[] GetSnapshot() => _persistentTxs.GetSnapshot();
 
-        public void Broadcast(Transaction tx, bool isPersistent)
+        public bool Broadcast(Transaction tx, bool isPersistent)
         {
             if (isPersistent)
             {
-                StartBroadcast(tx);
-            }
-            else
-            {
-                BroadcastOnce(tx);
+                return StartBroadcast(tx);
             }
+
+            BroadcastOnce(tx);
+
+            return true;
         }
 
-        private void StartBroadcast(Transaction tx)
+        private bool StartBroadcast(Transaction tx)
         {
+            if (tx is null) return false;
+
+            bool txInserted = _persistentTxs.TryInsert(tx.Hash, tx.SupportsBlobs ? new LightTransaction(tx) : tx);
+
+            if (!txInserted) return false;
+
             // broadcast local tx only if MaxFeePerGas is not lower than configurable percent of current base fee
             // (70% by default). Otherwise only add to persistent txs and broadcast when tx will be ready for inclusion
             if (tx.MaxFeePerGas >= _baseFeeThreshold || tx.IsFree())
             {
                 NotifyPeersAboutLocalTx(tx);
+                return true;
             }
 
-            if (tx.Hash is not null)
-            {
-                _persistentTxs.TryInsert(tx.Hash, tx.SupportsBlobs ? new LightTransaction(tx) : tx);
-            }
+            return false;
```

Comment:
After this change, we will return `false` if tx was inserted, but is not close to base fee. It will be possible, that we return `false`, but tx will be included in the block in the future. We should decide on one of two options:

1. just return `txInserted` at the end of the method. Even if tx is not good enough to send to peers right now, but if we add it to persistent collection, we need to return `true`
2. move attempt to insert tx inside `if` statement checking price, so just `return _persistentTxs.TryInsert(tx.Hash, tx.SupportsBlobs ? new LightTransaction(tx) : tx);` inside `if` and `return false` at the end

It depends if we want to add local tx with fee lower than 70% of `baseFee` or reject it. I don't have a strong preference here, you can decide. If I would need to make this decision, I will probably go with option 1